controller_manager:
  ros__parameters:
    update_rate: 10 # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    # diffbot_base_controller removed (wheel/motor interfaces removed)
    # Group trajectory controller for the 3-DOF manipulator
    arm_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

# diffbot_base_controller configuration removed because wheels/motors/encoders are no longer handled by the hardware interface
# odom_frame_id: odom
# base_frame_id: base_link
# pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
# twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]

# open_loop: true
# enable_odom_tf: true

# cmd_vel_timeout: 0.5
# #publish_limited_velocity: true
# use_stamped_vel: false
# #velocity_rolling_window_size: 10

# # Velocity and acceleration limits
# # Whenever a min_* is unspecified, default to -max_*
# linear.x.has_velocity_limits: true
# linear.x.has_acceleration_limits: true
# linear.x.has_jerk_limits: false
# linear.x.max_velocity: 1.0
# linear.x.min_velocity: -1.0
# linear.x.max_acceleration: 1.0
# linear.x.max_jerk: 0.0
# linear.x.min_jerk: 0.0

# angular.z.has_velocity_limits: true
# angular.z.has_acceleration_limits: true
# angular.z.has_jerk_limits: false
# angular.z.max_velocity: 1.0
# angular.z.min_velocity: -1.0
# angular.z.max_acceleration: 1.0
# angular.z.min_acceleration: -1.0
# angular.z.max_jerk: 0.0
# angular.z.min_jerk: 0.0

arm_trajectory_controller:
  ros__parameters:
    type: joint_trajectory_controller/JointTrajectoryController

    joints:
      - linkk_1
      - link_2
      - link_3

    command_interfaces:
      - position

    state_interfaces:
      - position

    state_publish_rate: 25.0
    action_monitor_rate: 10.0
    interpolation_mode: "linear"
    allow_partial_joints_goal: false
